py_ballisticcalc/munition.py

```diff
@@ -11,78 +11,70 @@
+    :param sight_height: Vertical distance from center of bore line to center of sight line.
+    :param twist: Distance for barrel rifling to complete one complete turn.
+        Positive value => right-hand twist, negative value => left-hand twist.
+    :param zero_elevation: Angle of barrel relative to sight line when sight is set to "zero."
+        (Typically computed by ballistic Calculator.)
     """
     sight_height: [float, Distance] = field(default_factory=lambda: Set.Units.sight_height)
-    zero_distance: [float, Distance] = field(default_factory=lambda: Set.Units.distance)
     twist: [float, Distance] = field(default_factory=lambda: Set.Units.twist)
-    zero_look_angle: [float, Angular] = field(default_factory=lambda: Set.Units.angular)
+    zero_elevation: [float, Angular] = field(default_factory=lambda: Set.Units.angular)
 
     def __post_init__(self):
+        if not self.sight_height:
+            self.sight_height = 0
         if not self.twist:
             self.twist = 0
-        if not self.zero_look_angle:
-            self.zero_look_angle = 0
+        if not self.zero_elevation:
+            self.zero_elevation = 0
 
 
 @dataclass
 class Ammo(TypedUnits):
-    """Creates Ammo and Projectile properties"""
-
-    dm: 'DragModel'
-    length: [float, Distance] = field(default_factory=lambda: Set.Units.length)
+    """
+    :param dm: DragModel for projectile
+    :param mv: Muzzle Velocity
+    :param powder_temp: Baseline temperature that produces the given mv
+    :param temp_modifier: Change in velocity w temperature: fps per °C.
+        Can be computed with .calc_powder_sens().  Only applies if
+        Settings.USE_POWDER_SENSITIVITY = True.
+    """
+    dm: DragModel = field(default=None)
     mv: [float, Velocity] = field(default_factory=lambda: Set.Units.velocity)
+    powder_temp: [float, Temperature] = field(default_factory=lambda: Set.Units.temperature)
     temp_modifier: float = field(default=0)
-    powder_temp: [float, Temperature] = field(default_factory=lambda: Temperature.Celsius(15))
+
+    def __post_init__(self):
+        if not self.powder_temp:
+            self.powder_temp = Temperature.Celsius(15)
 
     def calc_powder_sens(self, other_velocity: [float, Velocity],
                          other_temperature: [float, Temperature]) -> float:
-        """Calculates velocity correction by temperature change
-        :param other_velocity: other velocity
+        """Calculates velocity correction by temperature change; assigns to self.temp_modifier
+        :param other_velocity: other velocity at other_temperature
         :param other_temperature: other temperature
         :return: temperature modifier
         """
-        # (800-792) / (15 - 0) * (15/792) * 100 = 1.01
-        # creates temperature modifier in percent at each 15C
-        v0 = self.mv >> Velocity.MPS
+        v0 = self.mv >> Velocity.FPS
```

Comment:
What's the reason of changing the MPS to FPS here?

Reply:
I suspect I was thinking to match the units in TrajectoryCalc, but it's not necessary.